include/graaflib/algorithm/shortest_path.tpp

```diff
@@ -61,6 +61,80 @@ std::optional<GraphPath<int>> get_unweighted_shortest_path(
   }
 }
 
+template <typename V, typename E, graph_spec S>
+std::optional<GraphPath<E>> get_weighted_shortest_path(
+    const graph<V, E, S>& graph, vertex_id_t start_vertex,
+    vertex_id_t end_vertex) {
+  using weight_t = typename graph<V, E, S>::edge_t;
```

Comment:
Interestingly enough it seems like clang is confusing the name of the parameter and the type (which are both called `graaf`). Changing the parameter name works, but a slightly more elegant solution is:

```suggestion
    const graph<V, E, S>& graph, vertex_id_t start_vertex,
    vertex_id_t end_vertex) {
  using weight_t = typename decltype(graph)::edge_t;
```

I checked and this should work on both gcc as well as clang.